capstone-backend/src/main/java/com/google/sps/servlets/Utility.java

```diff
@@ -316,4 +319,114 @@ else if (("remove_"+field.getName()).equals(key)) {
+
+  /**
+   * This method assures that the given post request is valid, contains the necessary fields, and 
+   * does not contain keys that do not match object fields exactly.
+   * @param {jsonObject} the supplied request body in json form
+   * @param {response} response returned from the call
+   * @param {fields} the Fields (contains name and type) of the class being used  
+   * @param {requiredFields} a list of the names of fields deemed "necessary" for creation of an object.
+   * @return boolean true if it is a valid post request, false if not
+   */
+  public static <T extends BaseEntity> boolean postErrorHandler(JsonObject jsonObject, HttpServletResponse response, 
+      Field[] fields, List<String> requiredFields) throws IOException {
+    List<String> list = new ArrayList<>();
+    List<String> fieldNames = fields.stream().map(f -> f.getName()).collect(Collectors.toList());
+    list.addAll(jsonObject.keySet());
+    list.retainAll(fieldNames);
+    if (list.size() != jsonObject.keySet().size()) {
+      System.err.println("Error: Not all parameters in the request body are fields of the given class.");
+      response.sendError(HttpServletResponse.SC_BAD_REQUEST);
+      return false;
+    } 
+    list.clear();
+    list.addAll(jsonObject.keySet().stream()
+                    .filter(key -> jsonObject.get(key).getAsString().length() > 0)
+                    .collect(Collectors.toList()));
+    list.retainAll(requiredFields);
+    if (list.size() != requiredFields.size()) {
+      System.err.println("Error: Not all required fields are present in the request body, or some required fields are empty.");
+      response.sendError(HttpServletResponse.SC_BAD_REQUEST);
+      return false;
+    } 
+    return true;
+  }
+
+  /**
+   * This method executes the given post request, creating the new object and returning it with its ID appended.
+   * @param {collectionReference} reference to the appropriate database collection
+   * @param {request} the request sent to the backend
+   * @param {genericClass} a Generic class, to be used in casting objects retrieved from the database
+   * @param {requiredFields} a list of the names of fields deemed "necessary" for creation of an object.
+   * @return T the object created by the post request, with an ID field generated by firebase
+   */
+  public static <T extends BaseEntity> T post(CollectionReference collectionReference, HttpServletRequest request, 
+      HttpServletResponse response, GenericClass<T> genericClass, List<String> requiredFields) throws IOException {
+    Map<String, Object> constructorFields = new HashMap<>();
+    JsonObject jsonObject = Utility.createRequestBodyJson(request);
+    Field[] fields = genericClass.getMyType().getDeclaredFields();
+    if (!postErrorHandler(jsonObject, response, fields, requiredFields)) {
+      return null;
+    }
+    for (Field f : fields) {
+      String name = f.getName();
+      String type = f.getType().getName();
```

Comment:
nit: It might be more clear if you change this to fieldName or fName? "name" makes me think it's a field of an object, like the name of a BookList or the name of a User

Reply:
Makes sense, I'll change it for readability.